File: mobile/lib/features/product_services/domain/product_service_models.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'product_service_models.freezed.dart';
part 'product_service_models.g.dart';

@freezed
class ProductService with _$ProductService {
  const factory ProductService({
    required String id,
    required String workspaceId,
    required String brandId,
    required String name,
    required String description,
    required String category,
    required String targetAudience,
    required String sellingPoints,
    @Default('ACTIVE') String status,
    DateTime? createdAt,
  }) = _ProductService;

  factory ProductService.fromJson(Map<String, dynamic> json) => _$ProductServiceFromJson(json);
}

@freezed
class CreateProductServiceRequest with _$CreateProductServiceRequest {
  const factory CreateProductServiceRequest({
    required String brandId,
    required String name,
    required String description,
    required String category,
    required String targetAudience,
    required String sellingPoints,
  }) = _CreateProductServiceRequest;

  factory CreateProductServiceRequest.fromJson(Map<String, dynamic> json) => _$CreateProductServiceRequestFromJson(json);
}

File: frontend/src/app/features/product-services/product-service.models.ts
export type ProductServiceStatus = 'ACTIVE' | 'ARCHIVED';

export interface ProductServiceRecord {
  readonly id: string;
  readonly workspaceId: string;
  readonly brandId: string;
  readonly name: string;
  readonly description: string | null;
  readonly category: string | null;
  readonly targetAudience: string | null;
  readonly sellingPoints: string | null;
  readonly status: ProductServiceStatus;
  readonly createdAt: string;
  readonly updatedAt: string;
}

export interface CreateProductServicePayload {
  readonly name: string;
  readonly description: string | null;
  readonly category: string | null;
  readonly targetAudience: string | null;
  readonly sellingPoints: string | null;
}

export interface UpdateProductServicePayload extends CreateProductServicePayload {
  readonly status: ProductServiceStatus;
}
